### PR TITLE
feat(layer-cache): add get_bulk method to layer-cache

### PR DESCRIPTION
**Pull request by sprutton1**
This can be used to get many values from the cache at once. Values that fall through to database queries will be batched into a single query.

### CODE REVIEW

File: lib/si-layer-cache/src/pg.rs [hunk 1]
```diff
@@ -37,13 +39,14 @@ where
 
 impl<K> PgLayer<K>
 where
-    K: AsRef<[u8]> + Copy + Send + Sync,
+    K: AsRef<[u8]> + Copy + Send + Sync + std::hash::Hash + std::cmp::Eq,
```
Comment (zacharyhamm) [inline lib/si-layer-cache/src/pg.rs:42]: Did you end up needing these extra trait bounds? I don't expect we'll ever have a key that won't be hashable or support equality, but I think these were for the FromSql stuff?

Comment (sprutton1) [reply lib/si-layer-cache/src/pg.rs:42]: Both are required to insert into the hashmap on line 90 in `pg.rs`. Should I propagate the traits bounds to the other places we define k?

**Review by nickgerace — APPROVED** [review]

**Comment by sprutton1** [comment]
bors try

File: lib/si-layer-cache/src/pg.rs (hunk 1 above)
Comment (adamhjk) [reply lib/si-layer-cache/src/pg.rs]: You won't need them once our code merges, since K will always be an &str or String.

**Comment by sprutton1** [comment]
bors merge

**Pull request merged**